samples/CosmosDBIntegration/ReadThroughSamples/Models/StreamDataSingleDocument.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Extensions.Logging;
+using StackExchange.Redis;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Redis.Samples.CosmosDB.Models
+{
+    public class StreamDataSingleDocument
+    {
+        public string id { get; set; }
+        public int maxlen { get; set; }
+        public Dictionary<string, Dictionary<string, string>> messages { get; set; }
+
+        public static StreamDataSingleDocument CreateNewEntry(StreamEntry entry, string streamName, ILogger logger)
+        {
+            logger.LogInformation("Creating a new document for {val}. Inserting ID: {val} as the first entry", streamName, entry.Id.ToString());
+
+            // Map each key value pair
+            Dictionary<string, string> dict = entry.Values.ToDictionary(value => value.Name.ToString(), value => value.Value.ToString());
+
+            // Create a new list of messages
+            var list = new Dictionary<string, Dictionary<string, string>>();
```

Comment:
Can simplify to `            var list = new Dictionary<string, Dictionary<string, string>>
            {
                { entry.Id.ToString(), dict }
            };`